module-audio/Audio/Profiles/ProfileConfigUtils.cpp

```diff
@@ -0,0 +1,108 @@
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "ProfileConfigUtils.hpp"
+#include <fstream>
+#include <log/log.hpp>
+#include <json11.hpp>
+#include <magic_enum.hpp>
+namespace audio
+{
+    namespace strings
+    {
+        constexpr inline auto samplerate   = "samplerate";
+        constexpr inline auto bitWidth     = "bitWidth";
+        constexpr inline auto flags        = "flags";
+        constexpr inline auto outputVolume = "outputVolume";
+        constexpr inline auto outputPath   = "outputPath";
+        constexpr inline auto inputGain    = "inputGain";
+        constexpr inline auto inputPath    = "inputPath";
+        constexpr inline auto filterParams = "filterParams";
+        constexpr inline auto filterType   = "filterType";
+        constexpr inline auto frequency    = "frequency";
+        constexpr inline auto Q            = "Q";
+        constexpr inline auto gain         = "gain";
+    } // namespace strings
+
+    namespace utils
+    {
+
+        template <typename E, typename T>
+        constexpr inline typename std::enable_if<std::is_enum<E>::value && std::is_integral<T>::value, E>::type toEnum(
+            T value) noexcept
+        {
+            return static_cast<E>(value);
+        }
+
+        equalizer::FilterType toFilterType(const std::string &filterName)
+        {
+            auto filterType = magic_enum::enum_cast<equalizer::FilterType>(filterName);
+            if (filterType.has_value()) {
+                return filterType.value();
+            }
+            else {
+                LOG_ERROR("Unknown filter type, using none");
+                return equalizer::FilterType::None;
+            }
+        }
+
+        const std::string readFileToString(std::filesystem::path filePath)
+        {
+            std::ifstream file;
+            std::string configString;
+            LOG_DEBUG("Reading %s ...", filePath.c_str());
+            file.open(filePath);
+            if (not file.is_open()) {
+                LOG_ERROR("Can't open profile configuration file, using defaults!");
+                return std::string{};
+            }
+            while (file) {
+                std::string line;
+                std::getline(file, line);
+                configString += line;
+            }
+            file.close();
+            return configString;
+        }
+    } // namespace utils
+
+    const audio::codec::Configuration loadConfigurationFromFile(std::filesystem::path filePath)
+    {
+
+        auto configString = utils::readFileToString(filePath);
+        audio::codec::Configuration config;
+
+        json11::Json configJson;
+        std::string err;
+        configJson = json11::Json::parse(configString.c_str(), err);
+        if (!err.empty()) {
+            LOG_ERROR("Failed parsing device string!");
+            return config;
+        }
+        config.sampleRate_Hz = configJson[strings::samplerate].int_value();
```

Comment:
from the documentation of library used: 
```
There are also implicit constructors that allow standard and user-defined types to be automatically converted to JSON. 
```
Why wont we use it? :)

Reply:
TBH it wouldn't clarify it too much - I'd have to write a `toJson` function, which would look just the same :P
But sure, if it would be better I can rewrite it. The conversion is not used anywhere besides the profile constructor

Reply:
actually, there are some gains:

layout wise:
1. you will have a specialized class you test.
2. with a specialized class - you can always make a base class, and then create separate specialization if you wish for some other config loaded (i.e. from settings :) ) with the same values, just different loading
3. then you are able to test just derived classes parsing

by use:
1. you can then assign in the constructor and fail early. We cant parse? we throw in ctor and handle the exception in the layer below :) 

TL;DR;
I would say that it's a construction of proper config. I might try to argue it by this: https://isocpp.org/wiki/faq/exceptions#ctors-can-throw or fail fast principle :)

Reply:
I did read the documentation once again - it is possible to serialize it with `toJson` method, but we need a deserialize here, which is not possible this way :(

Reply:
Moved the load to the ctor with fallback handling